File: api-php/models/employee.php
<?php

/**
* Employee
*/
class Employee {
    
    /**
    * @var integer
    */
    private $id;
    
    /**
    * @var string
    */
    private $name;
    
    /**
    * @var \DateTime
    */
    private $clockIn;
    
    /**
    * @var \DateTime
    */
    private $clockOut;
    
    /**
    * @var boolean
    */
    private $active;
    
    function __construct($name, $active) {
        $this->id = uniqid();
        $this->name = $name;
        $this->clockIn = null;
        $this->clockOut = null;
        $this->active = $active;
    }
    
    /**
    * Get id
    *
    * @return integer
    */
    public function getId() {
        return $this->id;
    }
    
    /**
    * Set id
    * @param integer $id
    *
    * @return Employee
    */
    public function setId($id) {
        $this->id = $id;
        return $this;
    }
    
    /**
    * Set name
    *
    * @param string $name
    *
    * @return Employee
    */
    public function setName($name) {
        $this->name = $name;
        
        return $this;
    }
    
    /**
    * Get name
    *
    * @return string
    */
    public function getName() {
        return $this->name;
    }
    
    /**
    * Set clockIn
    *
    * @param \DateTime $clockIn
    *
    * @return Employee
    */
    public function setClockIn($clockIn) {
        $this->clockIn = $clockIn;
        
        return $this;
    }
    
    /**
    * Get clockIn
    *
    * @return \DateTime
    */
    public function getClockIn() {
        return $this->clockIn;
    }
    
    /**
    * Set clockOut
    *
    * @param \DateTime $clockOut
    *
    * @return Employee
    */
    public function setClockOut($clockOut) {
        $this->clockOut = $clockOut;
        
        return $this;
    }
    
    /**
    * Get clockOut
    *
    * @return \DateTime
    */
    public function getClockOut() {
        return $this->clockOut;
    }
    
    /**
    * Set active
    *
    * @param boolean $active
    *
    * @return Employee
    */
    public function setActive($active) {
        $this->active = $active;
        
        return $this;
    }
    
    /**
    * Get active
    *
    * @return boolean
    */
    public function getActive() {
        return $this->active;
    }
}

File: api-java/src/main/java/com/example/api/technicaltest/domain/Employee.java
package com.example.api.technicaltest.domain;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document(collection = "employees")
public class Employee {

    @Id
    private String id;
    private String name;
    private Date clockIn;
    private Date clockOut;
    private boolean active;

    public Employee(String name, Date clockIn, Date clockOut, boolean active) {
        this.name = name;
        this.clockIn = clockIn;
        this.clockOut = clockOut;
        this.active = active;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setClockIn(Date clockIn) {
        this.clockIn = clockIn;
    }

    public void setClockOut(Date clockOut) {
        this.clockOut = clockOut;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Date getClockIn() {
        return clockIn;
    }

    public Date getClockOut() {
        return clockOut;
    }

    public boolean isActive() {
        return active;
    }
}
